universe/Universe.cpp

```diff
@@ -2738,8 +2733,8 @@ std::set<int> Universe::RecursiveDestroy(int object_id) {
         Destroy(object_id);
         retval.insert(object_id);
 
-    } else if (auto planet = std::dynamic_pointer_cast<Planet>(obj)) {
-        for (int building_id : planet->BuildingIDs()) {
+    } else if (auto obj_planet = std::dynamic_pointer_cast<Planet>(obj)) {
```

Comment:
why?

Reply:
warning for shadowed on line 2776 `auto planet = GetPlanet(building->PlanetID());`

Reply:
The scopes are separate, I think...

Reply:
I was not aware until this PR:  this variables scope (from assignment in conditional) lasts until the end of the full block, not just within the success portion.
Clearest example I've come across is http://en.cppreference.com/w/cpp/language/if (bottom of page), `x` is still in scope within the else
Could break to separate if blocks with early returns

Reply:
huh... good to know.